### PR TITLE
throttle participant notifications in the UI

### PR DESCRIPTION
**Pull request by mmaxim**
Also add a throttler function

### CODE REVIEW

File: go/libkb/util.go [hunk 1]
```diff
@@ -1210,3 +1210,55 @@ func JsonwStringArray(a []string) *jsonw.Wrapper {
 	}
 	return aj
 }
+
+var throttleBatchClock clockwork.Clock = clockwork.NewRealClock()
```
Comment (joshblum) [inline go/libkb/util.go]: shouldn't the clock be per `ThrottleBatch` call?

File: go/libkb/util.go [hunk 2]
```diff
@@ -1210,3 +1210,55 @@ func JsonwStringArray(a []string) *jsonw.Wrapper {
...
+					lock.Unlock()
+					creation(throttleBatchEmpty{})
+				case <-cancelCh:
+					return
+				}
+			}()
+		}
+	}
+	return creation, func() {
+		close(cancelCh)
```
Comment (joshblum) [inline go/libkb/util.go]: scared this may be called twice and panic

File: go/libkb/util.go [hunk 3]
```diff
@@ -1210,3 +1210,55 @@ func JsonwStringArray(a []string) *jsonw.Wrapper {
...
+var throttleBatchClock clockwork.Clock = clockwork.NewRealClock()
+
+type throttleBatchEmpty struct{}
+
+func isEmptyThrottleData(arg interface{}) bool {
+	_, ok := arg.(throttleBatchEmpty)
+	return ok
+}
+
+func ThrottleBatch(f func(interface{}), batcher func(interface{}, interface{}) interface{},
```
Comment (joshblum) [inline go/libkb/util.go]: ~inter~sadface

File: go/libkb/util.go (hunk 2 above)
Comment (joshblum) [reply go/libkb/util.go]: if you add more state around this, can throttler just become a struct? feels more intuitive that way

Comment (mmaxim) [reply go/libkb/util.go]: Made it just do anything on first call

Comment (joshblum) [reply go/libkb/util.go]: why not `sync.Once`?

Comment (mmaxim) [reply go/libkb/util.go]: I forgot it existed

**Pull request merged**